pkg/controller/helpers/mergepatch.go

```diff
@@ -0,0 +1,36 @@
+package helpers
+
+import (
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func GenerateMergePatch(original runtime.Object, modified runtime.Object) ([]byte, error) {
+	if original == nil {
```

Comment:
these are interface objects so plain nil comparison doesn't work. I think you can skip these checks